src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/vm/AzureVMsProvider.java

```diff
@@ -0,0 +1,295 @@
+
+        // Get existing virtual private network if specified
+        Optional<Network> optionalVirtualNetwork = options.map(Options::getSubnetId)
+                                                          .map(subnetId -> azureProviderUtils.searchVirtualNetworkByName(azureService,
+                                                                                                                         subnetId)
+                                                                                             .get());
+
+        // Prepare a new  security group (same for all VMs)
+        Creatable<NetworkSecurityGroup> creatableNetworkSecurityGroup = azureProviderNetworkingUtils.prepareProactiveNetworkSecurityGroup(azureService,
+                                                                                                                                          region,
+                                                                                                                                          resourceGroup,
+                                                                                                                                          createUniqueSecurityGroupName(instance.getTag()));
+
+        // Get existing security group if specified
+        Optional<NetworkSecurityGroup> optionalNetworkSecurityGroup = options.map(Options::getSecurityGroupNames)
+                                                                             .map(secGrpNames -> secGrpNames.get(0))
+                                                                             .map(secGrpName -> azureProviderUtils.searchNetworkSecurityGroupByName(azureService,
+                                                                                                                                                    secGrpName)
+                                                                                                                  .get());
+
+        // Get existing public IP address if specified
+        Optional<PublicIpAddress> optionalPublicIpAddress = options.map(Options::getPublicIpAddress)
+                                                                   .map(publicIpAddress -> azureProviderUtils.searchPublicIpAddressByIp(azureService,
+                                                                                                                                        publicIpAddress)
+                                                                                                             .get());
+
+        return IntStream.rangeClosed(1,
+                                     Integer.valueOf(Optional.ofNullable(instance.getNumber())
+                                                             .orElse(SINGLE_INSTANCE_NUMBER)))
+                        .mapToObj(instanceNumber -> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this lambda expression number of lines from 33 to at most 20. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1188)